library/modules/EventManager.cpp

```diff
@@ -586,10 +595,19 @@ static void manageJobCompletedEvent(color_ostream& out) {
         if ( job0.flags.bits.repeat || job0.completion_timer != 0 )
             continue;
 
-        for (auto &key_value : copy) {
-            EventHandler &handle = key_value.second;
+        new_jobs_completed.emplace_back(&job0);
+    }
+
+    for (df::job* job : new_jobs_completed_repeats) {
+        for (auto &[_,handle] : copy) {
+            DEBUG(log,out).print("calling handler for repeated job completed event\n");
+            handle.eventHandler(out, (void*) job);
+        }
```

Comment:
what does moving this logic gain us? the `prevjobs` structure cannot be modified by the handlers (though the jobs could potentially be deleted, which would cause crashes both in the original code and in the proposed change)

Reply:
Yeah, there might be no way around that. Somehow I'm able to hold a job open in the gm-editor with auto-update long after its completion though, when it's no longer findable in the linked list? I might do some experiments later to determine the lifetime of a job.

Reply:
Actually, I think it's fine - the jobs in `prevJobs` are actually cloned job structures (by `Job::cloneJobStruct()`) and so won't be deleted by the game.

Reply:
oh that's good - I thought I remembered something like that, but I didn't see it when I was scanning the code for the review. So, this part can be refactored or not. You can refactor it for consistency with the other handlers or revert this section for fewer local variables. I'm ok with either way.